src/SFML/Window/Android/WindowImplAndroid.cpp

```diff
@@ -492,6 +503,28 @@ int WindowImplAndroid::processMotionEvent(AInputEvent* inputEvent, ActivityState
 
             states.touchEvents[id] = touchMoved.position;
         }
+        else if (static_cast<std::uint32_t>(device) & AINPUT_SOURCE_JOYSTICK)
+        {
+            // There seems to be no direct mapping between input event and the ID of the device that
+            // caused the event. However, as the single input event contains all axii changes, it's possible
+            // to poll values for all axii in a single loop iteration.
+            //
+            // Additionally, some controllers such as the Xbox One controller will report triggers as
+            // negative/positive values on the single axis on Windows, while Android reports them
+            // on two separate axii.
+            const auto deviceId = AInputEvent_getDeviceId(inputEvent);
+            if (states.joystickStates.find(deviceId) == states.joystickStates.end())
+                return 1;
+
+            const float factor = 100.f; // SFML normalizes axis to the range <-100, 100> instead of <-1, 1>
+            auto&       axes   = states.joystickStates.at(deviceId).axes;
```

Comment:
Direct access should be fine here, as we just checked it above:

```suggestion
            auto&       axes   = states.joystickStates[deviceId].axes;
```